### ISSUE TITLE
Specifying exact elasticsearch version means installation fails.

### ISSUE THREAD

**Issue by robtesch** [laravel/homestead#1361]
Please note that the Homestead issue tracker is reserved for bug reports and enhancements. We are not always able to debug Vagrant, Provider or Operating System issues, but will do our best to help. Thank you!

## Versions
  - Vagrant: Vagrant 2.2.6
  - Provider: Virtualbox 6.0.14
  - Homestead: On most recent release branch (10.2.0)

**A lot of issues can be resolved by simply updating vagrant, provider or homestead.**
>**Note:** Virtualbox users, please upgrade to `6.x`. You'll need to download a newer version of [Virtualbox](https://www.virtualbox.org/wiki/Downloads).

## Host operating system
Windows

## Homestead.yaml
```yaml
---
box: laravel/homestead
ip: "192.168.10.10"
memory: 8192
cpus: 6
provider: virtualbox

features:
    - golang: true
    - webdriver: true
    - elasticsearch:
        version: "7.1.1"
    - mysql8: true

authorize: ~/.ssh/id_rsa.pub

keys:
    - ~/.ssh/id_rsa

folders:
    - map: C:\Laravel\quickleads.local
      to: /home/vagrant/Laravel/quickleads.local
    - map: C:\Laravel\phpmyadmin.local
      to: /home/vagrant/Laravel/phpmyadmin.local
    - map: C:\Laravel\v2-ttr.local
      to: /home/vagrant/Laravel/v2-ttr.local
    - map: C:\Laravel\bits_and_bobs
      to: /home/vagrant/Laravel/bits_and_bobs

sites:
    - map: quickleads.local
      to: /home/vagrant/Laravel/quickleads.local/public
      schedule: true
      php: "7.3"
    - map: phpmyadmin.local
      to: /home/vagrant/Laravel/phpmyadmin.local
      php: "7.3"
    - map: v2-ttr.local
      to: /home/vagrant/Laravel/v2-ttr.local/public
      schedule: true
      php: "7.3"
    - map: localhost.v2-ttr.com
      to: /home/vagrant/Laravel/v2-ttr.local/public
      php: "7.3"
      
databases:
    - quickleads.local
    - v2ttr
    - v2ttr.testing

ports:
    - send: 9200
      to: 9200
    - send: 1323
      to: 1323
```

## Vagrant destroy & up output
Not available, as I've reverted to using less strict ES version for now.

## Expected behavior
When specifying version: "7.1.1" for elasticsearch, homestead should install exactly that version.

## Actual behavior
Installation failed at the point of trying to start elasticsearch with a "not found" error.
Note: if using version: 7 (less strict) then installation works without issue.

## Steps to reproduce
1. use homestead.yaml above.
2. vagrant destroy && vagrant up

## References
Nothing else i can find.

**Comment by browner12** [comment]
please follow the directions exactly.

https://laravel.com/docs/6.x/homestead#installing-optional-features

we only support major version selection.

**Comment by svpernova09** [comment]
What's interesting here is the v7 Repo is behaving differently than the v6 repo did. 

Part of the issue is in `scripts/features/elasticsearch.sh` on line 28:

```
echo "deb https://artifacts.elastic.co/packages/${version[0]}.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-${version[0]}.x.list
```

Should be updated to:

```
echo "deb https://artifacts.elastic.co/packages/7.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-7.x.list
```

This will hard code your script to only install versions available  7.x, if you need 6.x just undo this change. We'll need to get a PR to account for this particular change as well.

If you make this change and then `vagrant destroy && vagrant up` Elasticsearch will install version 7.1.1 as you'd expect:

Here are the available versions in the 7.x repo as of today (any of these versions will be installable via updating your `Homestead.yaml`

```
vagrant@homestead:~$ sudo apt-cache madison elasticsearch
elasticsearch |      7.5.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.5.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.5.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.4.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.4.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.4.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.3.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.3.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.3.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.2.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.2.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.1.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.1.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.0.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.0.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
```

This *should* solve your problem and get you back up and running. We'll need to sort out how we want to handle 6.x/7.x and get an update merged and new release to fix this for good.

**Comment by robtesch** [comment]
@browner12 

Sorry about that. I was reading the following documentation which is just below the section you mentioned:

```
Elasticsearch
You may specify a supported version of Elasticsearch, which may be a major version or an exact version number (major.minor.patch). The default installation will create a cluster named 'homestead'. You should never give Elasticsearch more than half of the operating system's memory, so make sure your Homestead machine has at least twice the Elasticsearch allocation.
```

**Comment by robtesch** [comment]
> What's interesting here is the v7 Repo is behaving differently than the v6 repo did.
> 
> Part of the issue is in `scripts/features/elasticsearch.sh` on line 28:
> 
> ```
> echo "deb https://artifacts.elastic.co/packages/${version[0]}.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-${version[0]}.x.list
> ```
> 
> Should be updated to:
> 
> ```
> echo "deb https://artifacts.elastic.co/packages/7.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-7.x.list
> ```
> 
> This will hard code your script to only install versions available 7.x, if you need 6.x just undo this change. We'll need to get a PR to account for this particular change as well.
> 
> If you make this change and then `vagrant destroy && vagrant up` Elasticsearch will install version 7.1.1 as you'd expect:
> 
> Here are the available versions in the 7.x repo as of today (any of these versions will be installable via updating your `Homestead.yaml`
> 
> ```
> vagrant@homestead:~$ sudo apt-cache madison elasticsearch
> elasticsearch |      7.5.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.5.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.5.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.4.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.4.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.4.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.3.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.3.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.3.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.2.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.2.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.1.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.1.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.0.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> elasticsearch |      7.0.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
> ```
> 
> This _should_ solve your problem and get you back up and running. We'll need to sort out how we want to handle 6.x/7.x and get an update merged and new release to fix this for good.

Thanks very much for looking at this. I thought about tinkering with that script but i'm no bash expert. Will make your changes are report back asap.

**Comment by svpernova09** [comment]
> @browner12
> 
> Sorry about that. I was reading the following documentation which is just below the section you mentioned:

What's funny is this is the first time I've seen someone try this w/ v7. :D

ES is the magic special feature that works differently than all the others because we were trying to support v6.0.0 which is what AWS at one time defaulted to, as well as supporting the latest at the time which I believe was ~6.5.

**Comment by robtesch** [comment]
> > @browner12
> > Sorry about that. I was reading the following documentation which is just below the section you mentioned:
> 
> What's funny is this is the first time I've seen someone try this w/ v7. :D
> 
> ES is the magic special feature that works differently than all the others because we were trying to support v6.0.0 which is what AWS at one time defaulted to, as well as supporting the latest at the time which I believe was ~6.5.

It's a real pain because AWS currently only supports up to 7.1, so i want to mirror the production environment as best as possible (hence i need 7.1, not the lastest version).

**Comment by svpernova09** [comment]
> It's a real pain because AWS currently only supports up to 7.1, so i want to mirror the production environment as best as possible (hence i need 7.1, not the lastest version).

Ah! Good to know. thanks for the info!

**Comment by browner12** [comment]
yah, looks like I was wrong on this one.  my bad.

I still remember back in the day when I PR'ed this in for ES5...

**Comment by svpernova09** [comment]
> yah, looks like I was wrong on this one. my bad.
> 
> I still remember back in the day when I PR'ed this in for ES5...

ye olde off by one errors :D You were mostly right.

**Comment by robtesch** [comment]
@svpernova09 I finally got round to trying your suggestion. Changed homestead.yaml as follows:

```
features:
    - golang: true
    - webdriver: true
    - elasticsearch:
        version: "7.1.1"
    - mysql8: true
```

Then changed line 28 in the elasticsearch.sh file as follows:
```
echo "deb https://artifacts.elastic.co/packages/7.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-7.x.list
```

The good news is, it no longer fails to install. The bad news is, it just goes ahead and installs the latest version, ignoring the 7.1.1 constraint.

I've also tried specifying the version like so (omitting the quotes) but sadly that made no difference:
```
version: 7.1.1
```

I am left with just changing line 31 in elasticsearch.sh to:
```
sudo apt-get -y install elasticsearch=7.1.1
```

This does work, but obviously means i have a very specific install script. I would try and dig into this more, but sadly my yaml parsing / bash skills are just not quite up to standard. Wish i could be of more help in providing a proper "fix".

**Comment by svpernova09** [comment]
I'm merging #1365 into master, but not tagging a release, could you test drive these changes on master and let me know if we're good?

To install the latest version of ES:

```
features:
    -   elasticsearch: true
```

To install version 6.2.2

```
features:
    -   elasticsearch:
            version: "6.2.2"
```

To install version 7.1.1

```
features:
    -   elasticsearch:
            version: "7.1.1"
```

Any of these versions are valid to be installed:

```
elasticsearch |      7.6.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.5.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.5.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.5.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.4.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.4.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.4.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.3.2 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.3.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.3.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.2.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.2.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.1.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.1.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.0.1 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      7.0.0 | https://artifacts.elastic.co/packages/7.x/apt stable/main amd64 Packages
elasticsearch |      6.8.6 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.6 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.8.5 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.5 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.8.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.8.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.8.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.8.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.8.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.8.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.7.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.7.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.7.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.7.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.7.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.7.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.6.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.6.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.6.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.6.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.6.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.6.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.5.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.5.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.5.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.5.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.5.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.5.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.5.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.5.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.5.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.5.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.4.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.4.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.4.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.4.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.4.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.4.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.4.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.4.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.3.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.3.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.3.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.3.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.3.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.3.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.2.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.2.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.2.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.2.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.2.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.2.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.2.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.2.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.2.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.2.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.1.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.1.4 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.1.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.1.3 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.1.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.1.2 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.1.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.1.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.1.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.1.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.0.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.0.1 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
elasticsearch |      6.0.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main amd64 Packages
elasticsearch |      6.0.0 | https://artifacts.elastic.co/packages/6.x/apt stable/main i386 Packages
```

**Comment by robtesch** [comment]
Will test drive today and let you know how it goes. Please give me a couple
of hours.

Thanks for your work on this!

**Comment by robtesch** [comment]
@svpernova09 looks like (for my particular case) it's solved! I haven't tried installing a specific ES6 version, but have tried with 7.1.1 and it did the job perfectly.

**Comment by svpernova09** [comment]
Resolved via #1365
